### PR TITLE
Implement sign/verify message as React components - Closes #352

### LINKED ISSUE #352: Migrate Sign/Verify message to React

### PR DESCRIPTION
**Pull request by slaweet**

### CODE REVIEW

**Comment by reyraa** [comment]
I see the In Progress label, but since I think this PR needs some changes I'd rather comment now to speed up the process.

File: src/app.js [hunk 1]
```diff
@@ -65,6 +72,9 @@ export default class App extends React.Component {
             <Route exact path="/" render={() => <p>Home</p>} />
           </div>
         </Router>
+        <Dialogs active={this.state.activeDialog}
```
Comment (reyraa) [inline src/app.js]: As mentioned above, better keeping App for higher level configs than specific components

File: package.json [hunk 2]
```diff
@@ -81,6 +83,7 @@
     "karma-webpack": "=2.0.3",
     "mocha": "=3.2.0",
     "postcss-loader": "=2.0.6",
+    "postcss-nested": "=2.0.2",
```
Comment (yasharAyari) [inline package.json]: we have already used cssnext plugin in our postcss config and it has nesting feature.So,there is no reason to use another plugin to add nesting feature to project.Please remove 'postcss-nested' and use cssnext nesting feature instead of that.I suggest to read cssnext features again.You can find more information here about nesting in cssnext in this link: http://cssnext.io/features/#nesting

File: src/components/dialogs/dialogs.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (reyraa) [inline src/components/dialogs/dialogs.js]: - If you please rename this file to `index.js`, we'll be able to import it with its directory: `src/components/dialogs`.
 - Also this can be in a separated PR as it's a shared component.

File: src/components/dialogs/dialogs.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { IconButton } from 'react-toolbox/lib/button';
+import Navigation from 'react-toolbox/lib/navigation';
+
+import VerifyMessage from '../signVerify/verifyMessage';
+import SignMessage from '../signVerify/signMessage';
+import styles from './dialogs.css';
+
+const Dialogs = props => (
+  <div>
+    {[
```
Comment (reyraa) [inline src/components/dialogs/dialogs.js]: The name of this file implies it's a generic component to handle any kind of child component. having configurations for *verify-message* and *sign-message* means we'll need the same configurations for all the components we may need later, yielding a long list of configurations. Please try writing a component that we can use it passing some properties.

File: package.json (hunk 2 above)
Comment (slaweet) [inline package.json]: I tried to remove this plugin and it broke my styles for sign/verify message dialog.
broken-css

Comment (yasharAyari) [inline package.json]: can you tell me which part of your style won't work ?

File: src/components/signVerify/signMessage.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      result: '',
+    };
+  }
+
+  handleChange(message) {
+    this.sign(message);
+  }
+
+  sign(message) {
+    const signnedMessage = lisk.crypto.signMessageWithSecret(message,
```
Comment (reyraa) [inline src/components/signVerify/signMessage.js]: Please fix the typo: signedMessage

File: src/components/signVerify/signMessage.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      this.props.account.passphrase);
+    const result = lisk.crypto.printSignedMessage(
+      message, signnedMessage, this.props.account.publicKey);
+    this.setState(Object.assign({}, this.state, { result, resultIsShown: false, message }));
+  }
+
+  showResult() {
+    if (this.state.result) {
+      const coppied = copy(this.state.result, {
```
Comment (reyraa) [inline src/components/signVerify/signMessage.js]: Please fix the typo: copied
There are multiple cases of this word.

File: src/components/signVerify/signMessage.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        debug: true,
+        message: 'Press #{key} to copy',
+      });
+      if (coppied) {
+        // TODO: set up the toaster in redux
+        // https://github.com/diegoddox/react-redux-toastr
+        toastr.success('Result copied to clipboard');
+      }
+      this.setState(Object.assign({}, this.state, { resultIsShown: true }));
```
Comment (reyraa) [inline src/components/signVerify/signMessage.js]: Please remove the`Object.assign` usage.

File: src/components/signVerify/signMessage.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                disabled={!this.state.result || this.state.resultIsShown}
+                onClick={this.showResult.bind(this)}/>
+          </section>
+          }
+      </div>
+    );
+  }
+}
+
+SignMessage.propTypes = {
```
Comment (reyraa) [inline src/components/signVerify/signMessage.js]: We're using `react-redux` library to implicitly pass propTypes.

File: src/components/signVerify/verifyMessage.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+      signature: {
+        error: '',
+        value: '',
+      },
+      result: '',
+    };
+  }
+
+  handleChange(name, value) {
+    const newState = Object.assign({}, this.state);
```
Comment (reyraa) [inline src/components/signVerify/verifyMessage.js]: I'd suggest changing this part to:
```
setState({ [name]: value });
```
and eliminating `Object.assing`

File: src/components/login/index.js [hunk 10]
```diff
@@ -1,7 +1,20 @@
 import React from 'react';
+import { Button } from 'react-toolbox/lib/button';
```
Comment (reyraa) [inline src/components/login/index.js]: These changes are not related to the ticket title. please remove.

File: src/components/header/index.js [hunk 11]
```diff
@@ -3,24 +3,48 @@ import { Button } from 'react-toolbox/lib/button';
 import { IconMenu, MenuItem } from 'react-toolbox/lib/menu';
 import logo from '../../assets/images/LISK-nano.png';
 import styles from './header.css';
+import store from '../../store';
```
Comment (reyraa) [inline src/components/header/index.js]: Please use `react-redux` connect method instead

**Comment by reyraa** [comment]
And also I appreciate the great job! thanks
